students/msirisha/lesson4/mailroom2.py

```diff
@@ -0,0 +1,61 @@
+# Mailroom script
+# part1. send a thank you note to donors.
+# Use the concepts learned so far.
+
+#global data structure
+donor_data = { "sai emani": {"first_name" : "sai", "last_name" : "emani", "donations" : [20.23, 30.456, 50.786]},
+               "sirisha marthy": {"first_name" : "sirisha", "last_name" : "marthy", "donations" : [67.89, 45.89]},
+               "ani emani": {"first_name" : "ani", "last_name" : "emani", "donations" : [12.789, 5.456]},
+               "charles dickens" : { "first_name":  "charles", "last_name": "Dickens", "donations": [15.89, 89.20, 345.67]},
+               "mark twain": {"first_name": "mark","last_name": "twain", "donations": [678.986]}}
```

Comment:
I might suggest just building a `dict` without the extra full name key, or you could do without individual name keys.  for example:
```
{
    "sai emani": [20.23, 30.456, 50.786],
    "sirisha marthy": [67.89, 45.89],
}
```
If you need the `first_name` and `last_name` you can do a `split(' ')` on the space between them.

```
for name, donations in donar_data.items():
    first_name = name.split(' ')[0]
    last_name = name.split(' ')[1]
```